### PR TITLE
[Feature] History Section

### PR DESCRIPTION
**Pull request by falbas**

### CODE REVIEW

File: src/components/modules/HistorySection/HistorySection.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      title: 'Lomba 3',
+      organizer: 'Universitas Brawijaya',
+      img: 'https://cdn.discordapp.com/attachments/826088171212636211/959083610831474729/compfest13-color_1.jpg',
+    },
+  ]
+
+  const handleNextSlide = useCallback(() => {
+    setActiveStyle('translate-x-[-200px] translate-y-[50px]')
+    setNextStyle('z-10')
+    setPrevStyle('translate-x-[200px] translate-y-[50px]')
```
Comment (gregoriusjimmy) [inline src/components/modules/HistorySection/HistorySection.tsx]: ini gabisa pakai conditional di classnamenya aja han? jadi statenya cuman perlu isActive, setIsActive

Comment (falbas) [reply src/components/modules/HistorySection/HistorySection.tsx]: nanti buat ngatur delaynya gimana ka?

Comment (falbas) [reply src/components/modules/HistorySection/HistorySection.tsx]: kalo gini gimana?

```
const handleNextSlide = useCallback(() => {
  setActive({ index: active.index, style: 'translate-x-[-200px] translate-y-[50px]' })
  setNext({ index: next.index, style: 'z-10' })
  setPrev({ index: prev.index, style: 'translate-x-[200px] translate-y-[50px]' })
  setTimeout(() => {
    setActive({ index: (active.index + 1) % 3, style: 'z-10' })
    setNext({ index: (next.index + 1) % 3, style: 'translate-x-[200px] translate-y-[50px]' })
    setPrev({ index: (prev.index + 1) % 3, style: 'translate-x-[-200px] translate-y-[50px]' })
  }, 300)
}, [active, next, prev])
```

Comment (gregoriusjimmy) [reply src/components/modules/HistorySection/HistorySection.tsx]: delaynya di setTimeout itu kan? mksd gw stylenya itu biar gadi taro di state langsung di classNamenya. posisi historynya diketahuin dari isActive  
 className={`absolute flex h-[300px] w-[300px] rounded-[20px] bg-white transition duration-500 ease-in-out ${
              index == active ? style aktif disini 
            } ${ :index == next ? next style disini }
            ${ :index == prev ? prev style disini }
            `}

**Comment by falbas** [comment]
bisa ni ka

```
const handleNextSlide = useCallback(() => {
  setActive((active + 1) % 3)
}, [active])
```

```
className={`absolute flex h-[300px] w-[300px] rounded-[20px] bg-white transition duration-500 ease-in-out ${
  index == active
    ? 'z-10'
    : index == (active + 1) % 3
    ? 'translate-x-[200px] translate-y-[50px]'
    : 'translate-x-[-200px] translate-y-[50px]'
}`}
```

**Comment by gregoriusjimmy** [comment]
> bisa ni ka
> 
> ```
> const handleNextSlide = useCallback(() => {
>   setActive((active + 1) % 3)
> }, [active])
> ```
> 
> ```
> className={`absolute flex h-[300px] w-[300px] rounded-[20px] bg-white transition duration-500 ease-in-out ${
>   index == active
>     ? 'z-10'
>     : index == (active + 1) % 3
>     ? 'translate-x-[200px] translate-y-[50px]'
>     : 'translate-x-[-200px] translate-y-[50px]'
> }`}
> ```

ntap lebih rapih kan, sbnernya cabang condition ? : yg nested kyak gitu kurang bagus karna hard to read, better pake &&. tapi gapapa nanti kita review bareng klo dah selesai apa aja yg bisa di improve also ini better daripada naruh style di state

**Review by gregoriusjimmy — APPROVED** [review]

**Pull request merged**